onnxscript/ir/_core.py

```diff
@@ -1977,9 +1977,86 @@ def insert_before(self, node: Node, new_nodes: Iterable[Node] | Node, /) -> None
         self._nodes.insert_before(node, new_nodes)
 
     def sort(self) -> None:
-        """Topologically sort the nodes in the graph."""
-        raise NotImplementedError("Not implemented yet")
-
+        # make sure the graph is a DAG
+        import heapq
+        stk = []
+        nodes = [node for node in self._nodes if isinstance(node, Node)] # screen nodes
```

Comment:
nodes is always a list of Node so the insinstance check is unnecessary.

```suggestion
```

Reply:
OK, but make sure all nodes are valid and belong to graph!